### PR TITLE
Upgraded nginx to 1.15.0

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
Signed-off-by: Jae Kim <jaebong.kim@gmail.com>

### CODE REVIEW

File: src/nginx/http.cc [hunk 1]
```diff
@@ -506,7 +506,7 @@ void wakeup_event_handler(ngx_event_t *ev) {
   // a nullptr pool definitely will cause crash.
   // Here, choose the least of evil, memory leak over crash.
   if (rp == nullptr || cp == nullptr) {
-    ngx_log_debug2(NGX_LOG_WARN_HTTP, ev->log, 0,
```
Comment (mangchiandjjoe) [inline src/nginx/http.cc]: NGX_LOG_WARN_HTTP has gone

**Comment by qiwzhang** [comment]
Could you run pr-e2e test in Jenkins?

File: src/nginx/module.cc [hunk 2]
```diff
@@ -1004,6 +1004,16 @@ ngx_int_t ngx_esp_init_module(ngx_cycle_t *cycle) {
   conf.module_type = NGX_HTTP_MODULE;
   conf.cmd_type = NGX_HTTP_MAIN_CONF;
 
+  // Fixed crash with v1.15.0
+  conf.conf_file =
+      (ngx_conf_file_t *)ngx_pcalloc(cycle->pool, sizeof(ngx_conf_file_t));
```
Comment (qiwzhang) [inline src/nginx/module.cc]: Seems like the code is not formatted properly.

**Comment by qiwzhang** [comment]
Could you rebase your changes

File: WORKSPACE [hunk 3]
```diff
@@ -28,7 +28,7 @@
 
 git_repository(
     name = "nginx",
-    commit = "18e870fcf52981cd0804eef269f2bb08182f0e3f",  # v1.13.4
+    commit = "2939d5c5fadb3f28a8f8ac85c039bfd007239d3c",  # v1.15.5
```
Comment (qiwzhang) [inline WORKSPACE]: can we update to the latest:  f5bf2d1

File: src/nginx/module.cc (hunk 2 above)
Comment (mangchiandjjoe) [reply src/nginx/module.cc]: Code was formatted using clang-format tool

File: WORKSPACE (hunk 3 above)
Comment (mangchiandjjoe) [reply WORKSPACE]: Done

**Comment by qiwzhang** [comment]
@mangchiandjjoe   Could you rebase your change?

File: third_party/BUILD [hunk 4]
```diff
@@ -44,16 +44,12 @@ nginx_suite(
...
-            "nginx-tests/limit_req.t",
-            # ssl2.t is excluded from the test suite because boring ssl
-            # does not support SSL_get0_raw_cipherlist used by $ssl_ciphers
-            # variable required by the test.
-            "nginx-tests/ssl2.t",
-            # ssl_sni_reneg.t is excluded from the test suite because boring ssl
-            # does not support renegotiation feature required by the test.
-            "nginx-tests/ssl_sni_reneg.t",
+            # ssl.t, ssl_certificates.t, ssl_stapling are excluded from the
+            # test suite because boring ssl does not support renegotiation
```
Comment (PiotrSikora) [inline third_party/BUILD]: `s/boring ssl/BoringSSL/`

Also, those tests are failing for reasons other than renegotiation, i.e. missing ciphers variable, missing support for multiple certificate types in the same context and missing support for OCSP stapling.

**Pull request merged**